apps/api/scripts/setup-competition.ts

```diff
@@ -40,21 +40,26 @@ const colors = {
 /**
  * Parse selection input like "1,3,5-8" into an array of indices
  */
-function parseSelectionInput(input: string, maxIndex: number): number[] {
+function parseSelectionInput(input: string, teams: Team[]): Team[] {
   try {
-    const result: number[] = [];
+    const maxIndex = teams.length;
+    const result: Team[] = [];
     const parts = input.split(",");
 
     for (const part of parts) {
       part.trim();
       if (part.includes("-")) {
         // Handle ranges (e.g., "5-8")
         const [start, end] = part.split("-").map((p) => parseInt(p.trim(), 10));
-        if (isNaN(start) || isNaN(end)) continue;
+        if (!start || !end || isNaN(start) || isNaN(end)) continue;
 
         for (let i = start; i <= end; i++) {
-          if (i > 0 && i <= maxIndex && !result.includes(i)) {
-            result.push(i);
+          if (
+            i > 0 &&
+            i <= maxIndex &&
+            !result.find((t) => t.id === teams[i - 1]!.id)
```

Comment:
Ugly, but a way to check that the team isn't already in the results. The set of selected teams hopefully isn't so huge that this causes performance problems. We can optimize later if needed by indexing teams or using a hashmap or whatever.